pkg/storage/stores/tsdb/single_file_index.go

```diff
@@ -335,12 +369,21 @@ func (i *TSDBIndex) SeriesVolume(ctx context.Context, _ string, from, through mo
 			}
 
 			if stats.Entries > 0 {
-				for _, l := range []labels.Label(ls) {
-					if _, ok := volumes[l.Name]; !ok {
-						volumes[l.Name] = make(map[string]uint64)
+				seriesLabels = seriesLabels[:0]
+				for _, l := range ls {
+					if _, ok := labelsToMatch[l.Name]; l.Name != TenantLabel && matchAll || ok {
```

Comment:
I think adding parens here would make this quicker to read. I understand this as `(l.Name != TenantLabel && matchAll) || ok`, is that correct?

Reply:
That makes sense.

Reply:
In my mind I was actually considering it to be more like `l.Name != TenantLabel && (matchAll || ok)`. Because things are resolved left-to-right, they all need to hold. The order here doesn't actually matter